src/backend/users/views.py

```diff
@@ -1,46 +1,60 @@
 from django.contrib.auth.models import User
 from django.core.mail import send_mail
+from django.db.utils import IntegrityError
 from django.http import HttpResponseRedirect
 from django.urls import reverse
 from rest_framework.authtoken.models import Token
+from rest_framework.exceptions import ValidationError
 from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
 from rest_framework.views import APIView
 from sesame import utils
 
-from .serializers import UserSerializer
+from .serializers import HouseSeekerSignUpSerializer, UserSerializer
+
+
+def send_login_link(request):
+    """
+    Send email with passwordless login link to User on Login/Sign Up
+    """
+    try:
+        # Request.data key could be "username" or "email", so turn into list to be safe
+        email = list(request.data.values())[0]
+        user = User.objects.get(username=email)
```

Comment:
I just had a thought: could we check for a user's profile and give them a blank one if it doesn't exist at this point? You're already doing that in the signup serializer, and I think it's a good idea to continue doing that, but doing it here would address the issue where someone who has been signed up manually using the admin or CLI may not have a profile set. There's no reason that we wouldn't want those people to be able to log in normally, so we might as well just give them a blank profile if we detect they're missing one.

Reply:
This is a great point! Actually, what do you think about creating a custom `User` manager class that creates an empty `UserProfile` instance with the `User` instance so everything's good to go on `create`? It might mean we have a build up of empty `UserProfile` instances, but like you said it would allow every `User` to be authorized to view the site by default once they have the link. I _think_ `Counselors` can only view/edit a `HouseSeeker` `UserProfile` in admin, but if not I can open an issue so the empty `UserProfiles` don't show up for them. Either way, I think a check for profile before sending the link is good here and will pop that in!

Reply:
Ok, got it! For now I'll implement the check/creating an empty `UserProfile` here and will keep an eye out for any funny business testing with admin/CLI.